File: app/src/main/java/org/mhsgv/portal/form/FormFactory.java
package org.mhsgv.portal.form;

public class FormFactory {

    public static FormFragment create(int formId) throws IllegalStateException {

        FormFragment form;
        switch(formId) {
            case 0:
                form = new Form1Fragment();
                break;
            case 1:
                form = new Form2Fragment();
                break;
            case 2:
                form = new Form3Fragment();
                break;
            default:
                throw new IllegalStateException("Haven't defined this one, yet.");
        }
        return form;
    }
}

File: app/src/main/java/org/mhsgv/portal/form/FormFactory.kt
package org.mhsgv.portal.form

object FormFactory {
    @JvmStatic
    @Throws(IllegalStateException::class)
    fun create(formId: Int): FormFragment {
        return when (formId) {
            0 -> Form1Fragment()
            1 -> Form2Fragment()
            2 -> Form3Fragment()
            else -> throw IllegalStateException("Haven't defined this one, yet.")
        }
    }
}